### PR TITLE
Max Chapter3/homework/4.

### PR DESCRIPTION
**Pull request by HellHunterMax**
Not finished but i need some help.
I'm stuck on a part where i cant add a specific school into a generic list of schools.

### CODE REVIEW

**Comment by Almantask** [comment]
refer to the interfaces below and use those instead. It simplifies the situation:
```cs
    // Leave it empty, because subjects are unrelated. Just for simulation
    // Alternatively use base Subject class.
    interface ISubject { }

    // For simulation you can store a specific teacher to school.
    // However for the interface based on requirements it is not needed.
    interface ISchool<TStudent> where TStudent : IStudent
    {
        // Missing:
        // Add
        // Get
    }

    interface IStudent
    {
        long Id { get; }

        void LearnFrom<TTeacher, TSubject>(TTeacher teacher)
            where TTeacher : ITeacher<TSubject>
            where TSubject : ISubject;
    }


    interface ITeacher<TSubject> where TSubject : ISubject
    {
        TSubject ProduceMaterial();
    }

```

**Comment by Almantask** [comment]
I'd recommend applying those new interfaces instead of trying working with the old ones ;)

**Comment by HellHunterMax** [comment]
i hope this is good enough?

File: Src/BootCamp.Chapter/SchoolSubjects/ISubject.cs [hunk 1]
```diff
@@ -6,7 +6,6 @@ namespace BootCamp.Chapter.SchoolSubjects
 {
     interface ISubject
```
Comment (Almantask) [inline Src/BootCamp.Chapter/SchoolSubjects/ISubject.cs:7]: can be removed

**Pull request closed without merging**